### PR TITLE
Issue 128

### PR DESCRIPTION
**Pull request by jverzani**
Close issue #128

### CODE REVIEW

File: src/context.jl [hunk 1]
```diff
@@ -137,13 +150,19 @@ end
 
 function _lookup_in_view(view::AbstractDict, key)
     ## is it a symbol?
+    k = occursin(r"^:", key) ? Symbol(key[2:end]) : key
```
Comment (GiggleLiu) [inline src/context.jl:153]:
```suggestion
    k = startswith(key, "^") ? Symbol(key[2:end]) : key
```

```julia
julia> using BenchmarkTools

julia> @benchmark startswith("~xxxx", "~")
BenchmarkTools.Trial: 
  memory estimate:  0 bytes
  allocs estimate:  0
  --------------
  minimum time:     5.394 ns (0.00% GC)
  median time:      5.647 ns (0.00% GC)
  mean time:        5.718 ns (0.00% GC)
  maximum time:     68.251 ns (0.00% GC)
  --------------
  samples:          10000
  evals/sample:     1000

julia> @benchmark occursin(r"^~", "~xxxx")
BenchmarkTools.Trial: 
  memory estimate:  0 bytes
  allocs estimate:  0
  --------------
  minimum time:     38.018 ns (0.00% GC)
  median time:      38.575 ns (0.00% GC)
  mean time:        38.885 ns (0.00% GC)
  maximum time:     110.374 ns (0.00% GC)
  --------------
  samples:          10000
  evals/sample:     992
```

File: src/context.jl [hunk 2]
```diff
@@ -34,54 +36,65 @@ function lookup_dotted(ctx::Context, dotted)
...
-            ## does name have a .?
-            if occursin(r"\.", key)
-                value = lookup_dotted(context, key)
-                value !== nothing && break
+        return ctx._cache[key]
+    end
+
+    # use global lookup down
+    global_lookup = false
+    if occursin(r"^~", key)
```
Comment (GiggleLiu) [inline src/context.jl:47]:
```suggestion
    if startswith(key, "^")
```

File: src/context.jl [hunk 3]
```diff
@@ -137,13 +150,19 @@ end
...
-    get(view, key, nothing)
+end
 
+# support legacy use of `first` and `second` as variable names
+# referring to piece, otherwise look up value
+function _lookup_in_view(view::Pair, key)
+    ## is it a symbol?
+    key == "first" && return view.first
+    key == "second" && return view.second
+    k =  occursin(r"^:", key) ?  Symbol(key[2:end]) : key
```
Comment (GiggleLiu) [inline src/context.jl:164]:
```suggestion
    k =  startswith(key, ":") ?  Symbol(key[2:end]) : key
```

**Comment by GiggleLiu** [comment]
Test coverage ↑↑↑↑↑

Thank you!

**Comment by jverzani** [comment]
> Test coverage ↑↑↑↑↑
> 
> Thank you!

Thank you!

**Pull request merged**